packages/jsx-email/src/renderer/render.ts

```diff
@@ -16,12 +21,41 @@ export const renderPlainText = async (
 ) => {
   const { formatters, selectors } = options || {};
 
-  const result = await jsxToString(component);
-  return htmlToText(result, {
+  // Generate HTML, then eagerly unwrap raw payloads using the same tolerant
+  // patterns as the HTML render path. This avoids edge cases where the HTML
+  // parser mutates the synthetic raw element closer and our formatter misses
+  // the payload. We still keep the formatter below as a fallback for any
+  // serialized text-node cases.
+  let result = await jsxToString(component);
+  result = result
+    .replace(RAW_ELEMENT_MARKER_UNWRAP_RE, (_whole, payload) => unescapeForRawComponent(payload))
+    .replace(RAW_ELEMENT_LEGACY_UNWRAP_RE, (_whole, payload) => unescapeForRawComponent(payload))
+    // Drop any remaining wrapper tags defensively (should be redundant once
+    // the unwrap runs, but helps when a serializer corrupts the closer).
+    .replace(/<\/?jsx-email-raw[^>]*?>/g, '');
+
+  // Protect closing-like raw sequences from being interpreted as HTML closers
+  // by the html-to-text parser. We'll restore them after conversion.
+  const LT_SLASH_SENTINEL = '__JSX_EMAIL_LT_SLASH__';
```

Comment:
the need for this is a sign that you have chosen a bad path and are over-engineering.